web/profiles/custom/yalesites_profile/modules/custom/ys_views_basic/src/ViewsBasicManager.php

```diff
@@ -0,0 +1,147 @@
+<?php
+
+namespace Drupal\ys_views_basic;
+
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\DependencyInjection\ContainerInjectionInterface;
+use Drupal\Core\Entity\EntityDisplayRepository;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+
+/**
+ * Service for managing the Views Basic plugins.
+ */
+class ViewsBasicManager extends ControllerBase implements ContainerInjectionInterface {
+
+  /**
+   * Allowed entity types for users to select.
+   *
+   * @todo This seems fragile and would better be inside a config page for
+   * admins to select.
+   *
+   * @var array
+   */
+  const ALLOWED_ENTITIES = [
+    'node' => 'node_type',
+    // 'media' => 'media_type',
+  ];
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The entity display repository.
+   *
+   * @var \Drupal\Core\Entity\EntityDisplayRepository
+   */
+  protected $entityDisplayRepository;
+
+  /**
+   * Constructs a new ViewsBasicManager object.
+   */
+  public function __construct(
+    EntityTypeManagerInterface $entity_type_manager,
+    EntityDisplayRepository $entity_display_repository
+    ) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->entityDisplayRepository = $entity_display_repository;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('entity_display.repository')
+    );
+  }
+
+  /**
+   * Returns an array of entity type machine names and the human readable name.
+   */
+  public function entityTypeList() {
+    $entityTypes = [];
+
+    foreach (self::ALLOWED_ENTITIES as $type) {
+      $types = $this->entityTypeManager()
+        ->getStorage($type)
+        ->loadMultiple();
+      foreach ($types as $machine_name => $object) {
+        $nodeType = $object->getEntityTypeId();
+        $value = "$nodeType.$machine_name";
+        $entityTypes[$value] = $object->label();
+      }
+    }
+
+    return $entityTypes;
+  }
+
+  /**
+   * Returns an array of view mode machine names and the human readable name.
+   */
+  public function viewModeList() {
+    $viewModes = [];
+
+    $view_modes = $this->entityTypeManager()
+      ->getStorage('entity_view_mode')
+      ->loadMultiple();
+    foreach ($view_modes as $machine_name => $object) {
+      foreach (self::ALLOWED_ENTITIES as $key => $type) {
+        $pattern = "/^{$key}/";
+        if (preg_match($pattern, $machine_name) && $object->status()) {
+          $viewModes[$machine_name] = $object->label();
+        }
+      }
+    }
```

Comment:
I'm glad both of these methods exist, but I think they will change significantly in the future. Your work is effective but likely does not reflect our actual business cases. We can create a separate ticket around "How you toy get the stuff in Drupal to fit neatly into the Views Basic UI?"

So far now let's roll with this. But the next version will need to be more curated for sure.

Reply:
For sure. I'm thinking something along the line of a `const` array or private property that will be specific entity ids with their associated view modes and "Human" readable names that make more sense to the content editors. For now, this is a good test of what we can get out of the back-end and into the front-end for the actual view and we can refactor later.